### PR TITLE
Caching moved to commons

### PR DESCRIPTION
**Pull request by vasichkin**

### CODE REVIEW

File: qubell/api/private/common.py [hunk 1]
```diff
@@ -20,7 +20,7 @@
 from qubell.api.tools import is_bson_id
 from qubell.api.private import exceptions
 from qubell import deprecated
-
+import pprint
```
Comment (dmakhno) [inline qubell/api/private/common.py]: remove

File: qubell/tests/common/test_cached.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import unittest
+from qubell.api.private.common import Cached
+import time
+import random
+
+class Stubs(object):
+    def respond_random(self):
+        return random.randint(1, 100)
+
+class test_Cached(unittest.TestCase):
```
Comment (dmakhno) [inline qubell/tests/common/test_cached.py]: class names in CamelCase

File: qubell/api/private/instance.py [hunk 3]
```diff
@@ -221,30 +220,8 @@ def technicalInfo(self): # This property is not always there. Waiting to appear
...
-        This allows to optimize calls when many parameters of entity requires withing short time.
-        """
-
-        if self.fresh():
-            return self.__cached_json
-        # noinspection PyAttributeOutsideInit
-        self.__last_read_time = time.time()
-        self.__cached_json = self._router.get_instance(org_id=self.organizationId, instance_id=self.instanceId).json()
-        return self.__cached_json
+        return Cached(data_fn=self._router.get_instance(org_id=self.organizationId, instance_id=self.instanceId).json)
```
Comment (dmakhno) [inline qubell/api/private/instance.py]: this is wrong

File: qubell/tests/common/test_cached.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import unittest
+from qubell.api.private.common import Cached
+import time
+import random
+
+class Stubs(object):
+    def respond_random(self):
+        return random.randint(1, 100)
```
Comment (dmakhno) [inline qubell/tests/common/test_cached.py]: this is flaky.
please introduce counter of this call, and assert the value of counter.

File: qubell/tests/common/test_cached.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    def setUp(self):
+        self.stubs = Stubs()
+
+    def test_response_cached(self):
+        resp = Cached(data_fn=lambda: self.stubs.respond_random())
+        query = resp.get()
+        assert resp.get() == query
+        assert resp.get() == query
+        time.sleep(3)
+        assert resp.get() != query
```
Comment (dmakhno) [inline qubell/tests/common/test_cached.py]: nope, in real life I still use `resp()`.

**Comment by dmakhno** [comment]
this won't work at all, each time I call `.json()` new empty Cached object will be returned.

File: qubell/tests/common/test_cached.py (hunk 4 above)
Comment (vasichkin) [inline qubell/tests/common/test_cached.py]: done

File: qubell/api/private/instance.py (hunk 3 above)
Comment (vasichkin) [inline qubell/api/private/instance.py]: Fixed, if i understand you right

File: qubell/tests/common/test_cached.py (hunk 2 above)
Comment (vasichkin) [inline qubell/tests/common/test_cached.py]: done

File: qubell/api/private/instance.py [hunk 6]
```diff
@@ -45,8 +45,8 @@ def __init__(self, organization, id):
         self.organization = organization
         self.organizationId = organization.organizationId
 
-        self.__cached_json = None
         self._last_workflow_started_time = None
+        self.__cached_json = Cached(data_fn=self._router.get_instance(org_id=self.organizationId, instance_id=self.instanceId).json)
```
Comment (dmakhno) [inline qubell/api/private/instance.py]: no way, this adds router call in constructor. this is not acceptable

**Pull request closed without merging**